modules/core/src/main/java/org/apache/fluo/core/impl/TransactionImpl.java

```diff
@@ -596,9 +596,17 @@ public CommitData createCommitData() {
 
   @Override
   public synchronized void commit() throws CommitException {
-    SyncCommitObserver sco = new SyncCommitObserver();
-    commitAsync(sco);
-    sco.waitForCommit();
+    SyncCommitObserver sco = null;
+    try {
+      sco = new SyncCommitObserver();
+      commitAsync(sco);
+      sco.waitForCommit();
+    } finally {
+      updates.clear();
+      weakNotification = null;
+      observedColumns.clear();
```

Comment:
No worries @keith-turner , thanks for looking at it.  Removed observedColumns.clear() allowed the ITs to pass for me locally.  We will see what travis has to say.  Thanks for opening that other ticket as well.